gapic/utils/lines.py

```diff
@@ -93,11 +93,12 @@ def wrap(text: str, width: int, *, offset: Optional[int] = None, indent: int = 0
                                 break_on_hyphens=False,
                                 )
         # Strip the first \n from the text so it is not misidentified as an
-        # intentionally short line below, except when the text contains `:`
-        # as the new line is required for lists.
+        # intentionally short line below, except when the text contains a list
+        # as the new line is required for lists. Look for `-` in the remaining
+        # text which indicates that a list is present.
```

Comment:
Do we need more sophisticated processing here? For example, what if there are list items with multiple paragraphs? Or non-list prose after the list? Or before?

 I'm assuming the parameter `text` can be pretty complex, like this:

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec porta euismod est a viverra. Integer vulputate ipsum id lacus tincidunt, id tincidunt tortor ullamcorper. Vestibulum facilisis at nulla nec lobortis. Nunc consectetur suscipit lacus id aliquam. 

Donec et urna aliquam, efficitur mauris et, consectetur enim. Aliquam aliquet turpis eget erat gravida condimentum. Sed vel feugiat risus. 

Sed interdum.

Convallis turpis nec congue. Integer vulputate sed urna eu mollis. Mauris in congue nisi, sed pellentesque ex.

- Ut vestibulum 
- consequat imperdiet
- Integer rhoncus varius 
  Ante, ac tempus augue finibus sit amet. Integer ac fermentum neque, a sodales nibh. Mauris et dictum ipsum. Integer sit amet posuere urna. Nullam cursus molestie posuere. Praesent imperdiet cursus purus, in posuere odio. 
- Orci varius natoque penatibus et 

Aagnis dis parturient montes, nascetur ridiculus mus. Mauris mattis turpis quis hendrerit gravida. Curabitur nec diam erat. In nec est nisl. Quisque ut orci efficitur, vestibulum ante non, vestibulum erat. Donec mollis ultricies nisl.
```

Reply:
Related to this, in line 84, we check for colon in `first` only. Should we not do this for every paragraph that is followed by a (sub-) list?

Reply:
Related:  http://shortn/_Nu6o0fKLzI `Bullet list ends without a blank line; unexpected unindent.` This is leading me to think we need more sophisticated parsing to prep things for Sphinx. Or we should indeed reconsider whether it's not worth making it an upstream requirement. @parthea

Reply:
I added a test for the sample that you provided in edef256 along with another fix to preserve blank lines.

Reply:
For the datacatalog docs failure, the issue was fixed upstream in protos here: cl/545771080. We need to update the public protos to resolve the failure.